libraries/logging/stream-logger-file/src/main/java/io/getstream/logging/file/FileStreamLogger.kt

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.logging.file
+
+import io.getstream.logging.Priority
+import io.getstream.logging.StreamLogger
+import io.getstream.logging.helper.stringify
+import java.io.BufferedWriter
+import java.io.Closeable
+import java.io.File
+import java.io.FileOutputStream
+import java.io.OutputStreamWriter
+import java.io.PrintWriter
+import java.io.Writer
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+
+private const val DEFAULT_SIZE: Int = 12 * 1024 * 1024
+
+private const val SHAREABLE_FILE: String = "stream_log_%s.txt"
+private const val INTERNAL_FILE_0: String = "internal_0.txt"
+private const val INTERNAL_FILE_1: String = "internal_1.txt"
+
+/**
+ * The [StreamLogger] implementation with log file persistence.
+ */
+public class FileStreamLogger(
+    private val config: Config,
+) : StreamLogger {
+
+    private val executor: Executor = Executors.newSingleThreadExecutor()
```

Comment:
Could we inject this collaborator by constructor?
On the other hand, could we move from "Threads" to "Coroutines"?

Reply:
I wanna keep this module clean from any 3rd party libs

Reply:
yes we can move it constructor

Reply:
We can use a default params with the `Executors.newSingleThreadExecutor()` value, but allowing to inject a different one on the case the one that will use this class needs it

Reply:
But I don't wanna allow passing any diff Executor by design to restrict it to be always a SingleThread one.